crates/solver/src/driver.rs

```diff
@@ -418,7 +418,10 @@ impl Driver {
                 .user_trades()
                 .map(|trade| {
                     let execution = Execution {
-                        surplus_fee: trade.surplus_fee(),
+                        // No need to save the surplus_fee for limit orders before the settlement is
+                        // settled onchain. This is done in the autopilot once the transaction is
+                        // mined.
+                        surplus_fee: None,
```

Comment:
Why is it ok to keep the solver_fee on there? Isn't this also derivable in the autopilot? I find this differentiation quite confusing and would prefer getting rid of persisting executions for unconfirmed transactions altogether.